src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -324,6 +340,87 @@ public ProxyProvider build() {
 		}
 	}
 
+	static final class RegexShouldProxyPredicate implements Predicate<SocketAddress> {
+
+		public static final RegexShouldProxyPredicate ALWAYS_PROXY = RegexShouldProxyPredicate.fromWildcardedPattern("");
+		public static final RegexShouldProxyPredicate DEFAULT_NON_PROXY = RegexShouldProxyPredicate.fromWildcardedPattern("localhost|127.*|[::1]|0.0.0.0|[::0]");
+
+		private final String regex;
+		private final Pattern pattern;
+
+		/**
+		 * Create a {@link RegexShouldProxyPredicate} based off the provided regular expression.
+		 *
+		 * @param regex The string regular expression
+		 * @return a predicate whether we should bypass the proxy
+		 */
+		public static RegexShouldProxyPredicate fromRegularExpression(String regex) {
+			return new RegexShouldProxyPredicate(regex);
+		}
+
+		/**
+		 * Creates a {@link RegexShouldProxyPredicate} based off the provided pattern with possible wildcards as
+		 * described in https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html
+		 *
+		 * @param pattern The string wildcarded expression
+		 * @return a predicate whether we should direct to proxy
+		 */
+		public static RegexShouldProxyPredicate fromWildcardedPattern(String pattern) {
+			String transformed;
+			if (StringUtil.isNullOrEmpty(pattern)) {
+				transformed = "$^"; // match nothing
+			}
+			else {
+				String[] parts = pattern.split("\\|");
+				for (int i = 0; i < parts.length; i++) {
+					parts[i] = transformWildcardComponent(parts[i]);
+				}
+				transformed = String.join("|", parts);
+			}
+			return new RegexShouldProxyPredicate(transformed);
+		}
+
+		private static String transformWildcardComponent(String in) {
+			String[] parts = new String[]{"", "", ""};
+			if (in.startsWith("*")) {
+				parts[0] = ".*";
+				in = in.substring(1);
+			}
+			if (in.endsWith("*")) {
+				parts[2] = ".*";
+				in = in.substring(0, in.length() - 1);
+			}
+			parts[1] = in;
```

Comment:
you should use `Pattern.quote` so that metacharacters or escape sequences in the input sequence will be given no special meaning.
Let's say we have something like this

```
@Test
public void nonProxyHosts_wildcardInTheMiddle() {
	ProxyProvider.RegexShouldProxyPredicate pred = ProxyProvider.RegexShouldProxyPredicate.fromWildcardedPattern("some.*.com");
	assertFalse("Should proxy, nothing matching foo", pred.test(someAddress("some.other.com")));
	assertFalse("Should proxy, nothing matching foo", pred.test(someAddress("some.foo.com")));
}
```

Reply:
good catch, fixed!